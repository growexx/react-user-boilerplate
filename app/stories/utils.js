import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import history from 'utils/history';
import configureStore from '../configureStore';
const store = configureStore({}, history);

export const WrappedStoryComponent = (Component, args = {}) => (
  <Provider store={store}>
    <IntlProvider locale="en">
      <ConnectedRouter history={history}>
        <Component {...args} />
      </ConnectedRouter>
    </IntlProvider>
  </Provider>
);
