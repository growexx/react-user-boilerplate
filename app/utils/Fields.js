/* eslint-disable react/prop-types */
/**
 * Redux-form Fields
 */
import React from 'react';
import { Form, Input, Radio, Select, Checkbox, DatePicker } from 'antd';

const FormItem = Form.Item;
const RadioGroup = Radio.Group;
const { TextArea } = Input;
const { RangePicker } = DatePicker;

const formItemLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
  },
};

const makeField = Component => ({
  input,
  meta,
  children,
  hasFeedback,
  label,
  ...rest
}) => {
  const hasError = meta.touched && meta.invalid;
  return (
    <FormItem
      {...formItemLayout}
      label={label}
      validateStatus={hasError ? 'error' : 'success'}
      hasFeedback={hasFeedback && hasError}
      help={hasError && meta.error}
    >
      <Component {...input} {...rest}>
        {children}
      </Component>
    </FormItem>
  );
};

export const AInput = makeField(Input);
export const APassword = makeField(Input.Password);
export const ARadioGroup = makeField(RadioGroup);
export const ASelect = makeField(Select);
export const ACheckbox = makeField(Checkbox);
export const ATextarea = makeField(TextArea);
export const ARangePicker = makeField(RangePicker);
