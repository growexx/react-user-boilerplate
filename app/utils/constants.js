export const RESTART_ON_REMOUNT = '@@saga-injector/restart-on-remount';
export const DAEMON = '@@saga-injector/daemon';
export const ONCE_TILL_UNMOUNT = '@@saga-injector/once-till-unmount';
export const TOKEN_KEY = 'jwtToken';
export const USER_DATA_KEY = 'userData';
export const EMITTER_EVENTS = {
  LOG_IN: 'USER_LOGGED_IN',
  LOG_OUT: 'USER_LOGGED_OUT',
};
export const GA_CATEGORY_MENU_CLICKS = 'Menu_Clicks';
