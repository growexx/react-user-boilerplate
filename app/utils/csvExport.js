import { Parser } from 'json2csv';

/**
 * @description Export Json data as CSV, on function call it'll convert
 * JSON data to blob and add phantom anchor for download and removes once download is complete
 * @param {Array} jsonData object array
 * @param {Object} csvOptions parser options
 * @param {String} fileName file name
 */
export const exportJsonAsCSV = (jsonData, csvOptions, fileName) => {
  try {
    const parser = new Parser(csvOptions);
    const csv = parser.parse(jsonData);
    const downloadLink = document.createElement('a');
    const blob = new Blob(['\ufeff', csv]);
    const url = URL.createObjectURL(blob);
    downloadLink.href = url;
    downloadLink.download = fileName;
    document.body.appendChild(downloadLink);
    downloadLink.click();
    document.body.removeChild(downloadLink);
    // eslint-disable-next-line no-empty
  } catch (err) {}
};
