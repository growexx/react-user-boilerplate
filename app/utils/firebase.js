import firebase from 'firebase';

// initializing firebase app
firebase.initializeApp({
  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.REACT_APP_FIREBASE_APP_ID,
});
export const auth = firebase.auth();
const googleProvider = new firebase.auth.GoogleAuthProvider();
const facebookProvider = new firebase.auth.FacebookAuthProvider();

export const signInWithGoogle = () => {
  auth
    .signInWithPopup(googleProvider)
    .then(res => {
      // eslint-disable-next-line no-console
      console.log(res.user);
    })
    .catch(error => {
      // eslint-disable-next-line no-console
      console.log(error.message);
    });
};

export const signInWithFacebook = () => {
  auth
    .signInWithPopup(facebookProvider)
    .then(res => {
      // eslint-disable-next-line no-console
      console.log(res.user);
    })
    .catch(error => {
      // eslint-disable-next-line no-console
      console.log(error.message);
    });
};
