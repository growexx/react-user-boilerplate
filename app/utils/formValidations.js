import validator from 'validator';

export const VALIDATION_MESSAGES = {
  EMAIL: 'Email is invalid!',
  URL: 'URL is invalid!',
  REQUIRED: 'Required!',
};
const isEmpty = value =>
  value === undefined || value === null || validator.isEmpty(value);
const join = rules => (value, data, props) =>
  rules
    .map(rule => rule(value, data, props))
    /* first error */
    .filter(error => !!error)[0];

/**
 * Email validation
 * @param(value)
 */

export function validEmail(value) {
  if (isEmpty(value) || !validator.isEmail(value)) {
    return VALIDATION_MESSAGES.EMAIL;
  }

  return '';
}

/**
 * isUrl validation
 *
 * @param(value)
 */

export function validUrl(value) {
  if (isEmpty(value) || !validator.isURL(value)) {
    return VALIDATION_MESSAGES.URL;
  }
  return false;
}

/**
 * required validation
 *
 * @param(value)
 */
export function required(value) {
  if (isEmpty(value)) {
    return VALIDATION_MESSAGES.REQUIRED;
  }
  return '';
}

/**
 * rules
 *
 * @param(rules).
 */

export function createValidator(rules) {
  return (data = {}, props) => {
    const errors = {};
    Object.keys(rules).forEach(key => {
      const rule = join([].concat(rules[key])); // concat enables both functions and arrays of functions
      const error = rule(data[key], data, props);
      if (error) {
        errors[key] = error;
      }
    });
    return errors;
  };
}

/* eslint-disable arrow-body-style */
