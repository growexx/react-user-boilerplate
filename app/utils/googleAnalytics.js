/**
 * REACT - GA : https://www.npmjs.com/package/react-ga
 */
import ReactGA from 'react-ga';
export const initGA = () => {
  ReactGA.initialize(process.env.REACT_APP_GA_TAG_ID, {
    testMode: process.env.NODE_ENV === 'test',
    titleCase: false,
  });
};

/**
 * recordPageView - record pages visited.
 * @param {string} location
 */
export const recordPageViewGA = location => {
  ReactGA.pageview(location);
};

/**
 * event - Add custom tracking event.
 * @param {string} category
 * @param {string} action
 * @param {string} label
 */
export const eventGA = (category, action, label) => {
  ReactGA.event({
    category,
    action,
    label,
  });
};
