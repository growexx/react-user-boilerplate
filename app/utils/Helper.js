import { TOKEN_KEY, USER_DATA_KEY } from './constants';
import StorageService from './StorageService';

export function userExists() {
  if (StorageService.exists(TOKEN_KEY)) {
    return true;
  }
  return false;
}

/**
 * Get User Data
 * @returns
 */
export function getUserData() {
  if (StorageService.exists(TOKEN_KEY)) {
    return StorageService.get(USER_DATA_KEY);
  }
  return false;
}

/**
 * to logout user from system
 */
export function logout() {
  StorageService.clear();
}

export function manageSession() {
  if (!localStorage.userData && !localStorage.loggedOut) {
    localStorage.setItem('loggedOut', 1);
    window.location.reload();
  } else if (localStorage.userData && !localStorage.loggedIn) {
    localStorage.setItem('loggedIn', 1);
    window.location.reload();
  }
}
