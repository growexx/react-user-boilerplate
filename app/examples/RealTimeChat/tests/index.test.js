/**
 *
 * Tests for RealTimeChat
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import 'jest-dom/extend-expect';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import history from 'utils/history';
import { browserHistory } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import configureStore from 'configureStore';
import RealTimeChat from 'examples/RealTimeChat/index';
import { getFireStoreCollectionReference } from 'utils/firebase';
import {
  chatWindowStub,
  getSuccessMockSearchResults,
  getFailureMockSearchResults,
} from 'examples/RealTimeChat/stub';
jest.mock('utils/firebase');
jest.mock('utils/Helper', () => ({
  getUserData: jest.fn().mockReturnValue({
    email: 'johnDoe@growexx.com',
  }),
}));
const mockFirebaseHelperFunctions = responseType => {
  getFireStoreCollectionReference.mockImplementation(() => ({
    where: jest.fn().mockImplementation(() => ({
      onSnapshot: jest.fn(),
      get: jest
        .fn()
        .mockImplementationOnce(() =>
          responseType === 'success'
            ? getSuccessMockSearchResults()
            : getFailureMockSearchResults(),
        ),
    })),
  }));
};
let store;
const props = {
  selectedChatWindow: chatWindowStub.joined,
};
const componentWrapper = () =>
  render(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ConnectedRouter history={history}>
          <RealTimeChat {...props} />
        </ConnectedRouter>
      </IntlProvider>
    </Provider>,
  );

describe('<RealTimeChat />', () => {
  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });
  afterEach(() => {
    getFireStoreCollectionReference.mockReset();
  });
  it('Should select the value from dropdown', async () => {
    mockFirebaseHelperFunctions('success');
    const { getByRole, debug, getByText } = componentWrapper();
    fireEvent.mouseDown(getByRole('combobox'));
    fireEvent.change(getByRole('combobox'), {
      target: {
        value: 'johndoe_9@gmail.com',
      },
    });
    debug();
    await waitForElement(() => getByRole('option'));
    fireEvent.click(
      document.querySelectorAll('.ant-select-item-option-content')[0],
    );
    expect(getByText('johndoe_9@gmail.com')).toBeInTheDocument();
  });
  it('Should render and match the snapshot', () => {
    mockFirebaseHelperFunctions('error');
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });
});
