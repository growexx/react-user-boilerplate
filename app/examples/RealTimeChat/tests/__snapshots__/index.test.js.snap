// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RealTimeChat /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="StyledRealTimeChat-sc-1iv8eur-0 ecjmtr"
  >
    <div
      class="searchContainer"
    >
      <div
        class="ant-select ant-select-auto-complete ant-select-single ant-select-allow-clear ant-select-show-search"
      >
        <div
          class="ant-select-selector"
        >
          <span
            class="ant-select-selection-search"
          >
            <input
              aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
              aria-autocomplete="list"
              aria-controls="rc_select_TEST_OR_SSR_list"
              aria-haspopup="listbox"
              aria-owns="rc_select_TEST_OR_SSR_list"
              autocomplete="off"
              class="ant-select-selection-search-input"
              id="rc_select_TEST_OR_SSR"
              role="combobox"
              type="search"
              value=""
            />
          </span>
          <span
            class="ant-select-selection-placeholder"
          >
            Search Users
          </span>
        </div>
      </div>
    </div>
    <div
      class="StyledRealTimeChat__ChatContainer-sc-1iv8eur-1 jlbzAe"
    >
      <div
        class="StyledChatList-z2vxes-0 dxvqQA"
      >
        <div
          class="StyledChatList__ChatListContainer-z2vxes-1 eECGnF"
        >
          <div
            class="demo-infinite-container"
          >
            <div>
              <div
                class="ant-list ant-list-split"
              >
                <div
                  class="ant-spin-nested-loading"
                >
                  <div
                    class="ant-spin-container"
                  >
                    <div
                      class="ant-list-empty-text"
                    >
                      <div
                        class="ant-empty ant-empty-normal"
                      >
                        <div
                          class="ant-empty-image"
                        >
                          <svg
                            class="ant-empty-img-simple"
                            height="41"
                            viewBox="0 0 64 41"
                            width="64"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fill-rule="evenodd"
                              transform="translate(0 1)"
                            >
                              <ellipse
                                class="ant-empty-img-simple-ellipse"
                                cx="32"
                                cy="33"
                                rx="32"
                                ry="7"
                              />
                              <g
                                class="ant-empty-img-simple-g"
                                fill-rule="nonzero"
                              >
                                <path
                                  d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                />
                                <path
                                  class="ant-empty-img-simple-path"
                                  d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                        <div
                          class="ant-empty-description"
                        >
                          No Data
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="StyledChatRoom-sc-1u84jil-0 cEReyG"
      >
        <div
          class="chatRoomContainer"
        >
          <div
            class="chatRoomHeader"
          >
            <p />
            <span
              aria-label="close"
              class="anticon anticon-close"
              role="img"
              tabindex="-1"
            >
              <svg
                aria-hidden="true"
                data-icon="close"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </span>
          </div>
          <div
            class="messageContainer"
          />
          <div
            class="sendMessageContainer"
          >
            <form
              class="ant-form ant-form-horizontal messageInput"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        placeholder="Enter Your Message"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="ant-btn ant-btn-primary"
                type="submit"
              >
                <span>
                  Send
                </span>
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
